js/src/HappoRunner.js

```diff
@@ -153,26 +154,33 @@ window.happo = {
     return document.body.children;
   },
 
+  /**
+   * @return {Promise}
+   */
   processExample(currentExample) {
-    const rootNodes = this.getRootNodes();
-    try {
-      const {
-        height,
-        left,
-        top,
-        width,
-      } = getFullRect(rootNodes);
-
-      return {
-        description: currentExample.description,
-        height,
-        left,
-        top,
-        width,
-      };
-    } catch (error) {
-      return handleError(currentExample, error);
-    }
+    return new Promise((resolve) => {
+      waitForImagesToRender().then(() => {
+        try {
+          const rootNodes = this.getRootNodes();
+          const {
+            height,
+            left,
+            top,
+            width,
+          } = getFullRect(rootNodes);
+
+          resolve({
+            description: currentExample.description,
+            height,
+            left,
+            top,
+            width,
+          });
+        } catch (error) {
+          resolve(handleError(currentExample, error));
```

Comment:
I guess, although we end up calling the `doneFunc` with an object that the ruby code inspects. This used to be a return so I just made it resolve.